task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -392,36 +393,50 @@ def _pending_event_processing_loop(self) -> None:
+            ),
+            resources=V1ResourceRequirements(
+                limits={
+                    "cpu": task_config.cpus,
+                    "memory": f"{task_config.memory}Mi",
+                    "ephemeral-storage": f"{task_config.disk}Mi",
+                }
+            ),
+            env=get_kubernetes_env_vars(
+                environment=task_config.environment,
+                secret_environment=task_config.secret_environment,
+                field_selector_environment=task_config.field_selector_environment,
+            ),
+            volume_mounts=get_kubernetes_volume_mounts(task_config.volumes),
+            ports=[V1ContainerPort(container_port=port) for port in task_config.ports],
+        )
+
     def run(self, task_config: KubernetesTaskConfig) -> Optional[str]:
         try:
-            container = V1Container(
-                image=task_config.image,
-                # XXX: we were initially planning on using the name from KubernetesTaskConfig here,
-                # but its too easy to go over the length limit for container names (63 characters),
-                # so we're just hardcoding something for now since container names aren't used for
-                # anything at the moment
-                name="main",
-                command=["/bin/sh", "-c"],
-                args=[task_config.command],
-                security_context=get_security_context_for_capabilities(
-                    cap_add=task_config.cap_add,
-                    cap_drop=task_config.cap_drop,
-                ),
-                resources=V1ResourceRequirements(
-                    limits={
-                        "cpu": task_config.cpus,
-                        "memory": f"{task_config.memory}Mi",
-                        "ephemeral-storage": f"{task_config.disk}Mi",
-                    }
-                ),
-                env=get_kubernetes_env_vars(
-                    environment=task_config.environment,
-                    secret_environment=task_config.secret_environment,
-                    field_selector_environment=task_config.field_selector_environment,
-                ),
-                volume_mounts=get_kubernetes_volume_mounts(task_config.volumes),
-                ports=[V1ContainerPort(container_port=port) for port in task_config.ports],
-            )
+            # XXX: we were initially planning on using the name from KubernetesTaskConfig here,
+            # but its too easy to go over the length limit for container names (63 characters),
+            # so we're just hardcoding something for now since container names aren't used for
+            # anything at the moment
+            containers = [self._create_container_definition("main", task_config)]
+
+            for name, nested_config in task_config.extra_containers.items():
+                containers.append(self._create_container_definition(
+                    get_sanitised_kubernetes_name(name),
```

Comment:
```suggestion
                    get_sanitised_kubernetes_name(name, length_limit=63),
```

https://github.com/Yelp/task_processing/blob/850c64cc6346237ccffbe622aceb0f469ff582cd/task_processing/plugins/kubernetes/utils.py#L105 has a default of no length limit (could also replace this 63 with a constant)